src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageLexer.g4

```diff
@@ -205,6 +205,7 @@ LITERAL_YIELD:           'yield';
 LITERAL_NON_SEALED:      'non-sealed';
 LITERAL_SEALED:          'sealed';
 LITERAL_PERMITS:         'permits';
+LITERAL_WHEN:            'when';
```

Comment:
Please add `when` to the pattern for illegal identifiers:
https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L789

Reply:
Ticket opened at https://github.com/checkstyle/checkstyle/issues/12553